lib/inspec_tools/summary.rb

```diff
@@ -115,12 +84,78 @@ def threshold_compliance
+    # rubocop:enable Metrics/PerceivedComplexity
+    # rubocop:enable Metrics/CyclomaticComplexity
+
+    private
+
+    def parse_threshold(threshold_inline, threshold_file)
+      threshold = Utils::InspecUtil.to_dotted_hash(YAML.load_file(THRESHOLD_TEMPLATE))
+      threshold.merge!(Utils::InspecUtil.to_dotted_hash(YAML.load_file(threshold_file))) if threshold_file
+      threshold.merge!(Utils::InspecUtil.to_dotted_hash(YAML.safe_load(threshold_inline))) if threshold_inline
+
+      return threshold
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def compute_summary
+      data = Utils::InspecUtil.parse_data_for_ckl(@json)
+
+      data.keys.each do |control_id|
+        current_control = data[control_id]
+        current_control[:compliance_status] = Utils::InspecUtil.control_status(current_control, true)
+        current_control[:finding_details] = Utils::InspecUtil.control_finding_details(current_control, current_control[:compliance_status])
+      end
+
+      summary = {}
+      summary[:buckets] = {}
+      summary[:buckets][:failed]    = select_by_status(data, 'Open')
+      summary[:buckets][:passed]    = select_by_status(data, 'NotAFinding')
+      summary[:buckets][:no_impact] = select_by_status(data, 'Not_Applicable')
+      summary[:buckets][:skipped]   = select_by_status(data, 'Not_Reviewed')
+      summary[:buckets][:error]     = select_by_status(data, 'Profile_Error')
+
+      summary[:status] = {}
+      summary[:status][:failed]    = tally_by_impact(summary[:buckets][:failed])
+      summary[:status][:passed]    = tally_by_impact(summary[:buckets][:passed])
+      summary[:status][:no_impact] = tally_by_impact(summary[:buckets][:no_impact])
+      summary[:status][:skipped]   = tally_by_impact(summary[:buckets][:skipped])
+      summary[:status][:error]     = tally_by_impact(summary[:buckets][:error])
+
+      summary[:compliance] = compute_compliance(summary)
+      summary
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    def select_by_impact(controls, impact)
+      controls.select { |_key, value| value[:impact].to_f.eql?(impact) }
+    end
+
+    def select_by_status(controls, status)
+      controls.select { |_key, value| value[:compliance_status].eql?(status) }
+    end
+
+    def tally_by_impact(controls)
+      tally = {}
+      tally[:total]    = controls.count
+      tally[:critical] = select_by_impact(controls, CRITICAL).count
+      tally[:high]     = select_by_impact(controls, HIGH).count
+      tally[:medium]   = select_by_impact(controls, MEDIUM).count
+      tally[:low]      = select_by_impact(controls, LOW).count
+      tally
+    end
 
-    def parse_threshold(new_threshold)
-      @threshold.merge!(new_threshold)
+    def compute_compliance(summary)
+      (summary[:status][:passed][:total]*100.0/
+        (summary[:status][:passed][:total]+
+         summary[:status][:failed][:total]+
+         summary[:status][:skipped][:total]+
+         summary[:status][:error][:total])).floor
```

Comment:
Is it possible to do something like `summary[:status].sum { |key| key[:total] }` here?